### PR TITLE
fix: add tooltip for long titles

### LINKED ISSUE #3673: Bug: Extended Workspace Names spill over dropdown.
### Describe the bug

Long titles are spilling outside of the sidebar. We should truncate them. 
image

### Steps to reproduce

1. Create a workspace with a long name.
2. Save the workspace
3. See the spillage.

### PR DESCRIPTION
**Pull request by SURAJ-SHARMA27**
## Description

Fixes #3673

This pull request fixes the issue where a long title would overflow from the side drawer. Now, the PR includes truncated labels for long titles, with the full label shown in a tooltip upon hovering. 

## Related Tickets & Documents

This PR fixes issue #3673
## Mobile & Desktop Screenshots/Recordings

[screen-capture (21).webm](https://github.com/user-attachments/assets/daba5015-5958-4acc-ac41-9cf5a0ec9952)

## Steps to QA
Test cases were not added as I didn't find the corresponding test file for this component. Should I write it from scratch?
1. Go to workspace
2. Create new workspace with a long name
3. See the truncated title and tooltip upon hovering

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

**Comment by jpmcb** [comment]
Accepted deploy preview and requested engineering's review: thanks for the contribution!!

File: components/atoms/Select/single-select.tsx [hunk 1]
```diff
@@ -45,6 +45,20 @@ const SingleSelect = ({
...
+    const words = label.split(" ");
+    let truncated = label;
+    if (words.length > wordLimit) {
+      truncated = words.slice(0, wordLimit).join(" ") + "...";
+    }
+    if (truncated.length > maxChars) {
+      truncated = truncated.slice(0, maxChars - 3) + "...";
+    }
+    return truncated;
+  };
```
Comment (OgDev-01) [inline components/atoms/Select/single-select.tsx:58]: Do we really need this??.

Same behaviour can be achieved with tailwinds truncate

File: components/atoms/Select/single-select.tsx [hunk 2]
```diff
@@ -63,7 +77,27 @@ const SingleSelect = ({
           insetLabel && `before:content-[attr(data-inset-label)] before:mr-1 before:font-normal before:text-slate-500`
         )}
       >
-        <p className="grow text-start">{current?.label ?? placeholder}</p>
+        {isTruncated ? (
+          <Tooltip.Provider>
```
Comment (OgDev-01) [inline components/atoms/Select/single-select.tsx:81]: we have a tooltip component already [here](https://github.com/open-sauced/app/blob/6781c1c010c1f763b5bffdf1b862c30d04f7ee6f/components/atoms/Tooltip/tooltip.tsx#L14) you can use that for consistency

**Review by OgDev-01 — COMMENTED** [review]
Hey @SURAJ-SHARMA27 ... Great job on patching this fix 👏 

Just a couple comments and pointers around the logic

File: components/atoms/Select/single-select.tsx (hunk 1 above)
Comment (SURAJ-SHARMA27) [reply components/atoms/Select/single-select.tsx]: yes i have replaced the function with tailwind truncate.

File: components/atoms/Select/single-select.tsx (hunk 2 above)
Comment (SURAJ-SHARMA27) [reply components/atoms/Select/single-select.tsx]: I have made this change also could you review it once?

image

File: components/atoms/Select/single-select.tsx [hunk 3]
```diff
@@ -63,7 +63,12 @@ const SingleSelect = ({
           insetLabel && `before:content-[attr(data-inset-label)] before:mr-1 before:font-normal before:text-slate-500`
         )}
       >
-        <p className="grow text-start">{current?.label ?? placeholder}</p>
+        <Tooltip content={current?.label ?? placeholder}>
+          <div className="flex items-center w-[200px]">
```
Comment (OgDev-01) [inline components/atoms/Select/single-select.tsx:67]: nit: we want to try as much as possible to reduce hard-coded values...
```suggestion
          <div className="flex items-center w-48">
```

Comment (SURAJ-SHARMA27) [reply components/atoms/Select/single-select.tsx]: Thank you @OgDev-01

**Review by brandonroberts — APPROVED** [review]
LGTM

**Review by jpmcb — APPROVED** [review]

**Pull request merged**